src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -182,11 +193,14 @@ public void updateGrades(@RequestBody List<GradeDTO> dlist) {
     // student lists their assignments/grades for an enrollment ordered by due date
     // student must be enrolled in the section
     @GetMapping("/assignments")
+    @PreAuthorize("hasAuthority('SCOPE_ROLE_STUDENT')")
     public List<AssignmentStudentDTO> getStudentAssignments(
-            @RequestParam("studentId") int studentId,
+            Principal principal,
             @RequestParam("year") int year,
             @RequestParam("semester") String semester) {
 
+        int studentId = userRepository.findByEmail(principal.getName()).getId();
+
```

Comment:
Adds @PreAuthorization tags to the methods requiring secure connections.
Converts @RequestParams to a principal call.